### PR TITLE
feat(release): sync-aware preview and standing PR rendering

### PR DESCRIPTION
**Pull request by goosewobbler**
## Summary

Fixes the contradictory package counts in the release preview under sync versioning. The preview for this repo currently says **"1 package queued"** (it counts changelogs — sync mode emits a single aggregated one) right above **"5 packages"** (it counts updates — which include the root lockstep bump). Neither number describes the actual release: 4 publishable packages moving as one unit.

## Approach

The formatter couldn't distinguish sync from async output, nor a root bump from a real package — neither fact was recorded. This PR records both in the `VersionOutput` contract (both optional, so old standing-PR manifests fall back to the previous rendering):

- `VersionOutput.strategy: 'sync' | 'single' | 'async'`
- `VersionPackageUpdate.isRoot` — marks the workspace-root lockstep bump

## What changes in sync mode

**Preview comment:**

> **Standing release PR:** #240 · v0.24.0 queued (4 packages) · open 13m · ✅ ready to merge
>
> **Release Preview** — v0.23.0 → v0.24.0
>
> **All packages → 0.24.0** (sync versioning)
> - `@releasekit/version`
> - `@releasekit/notes`
> - `@releasekit/publish`
> - `@releasekit/release`

**Standing PR:**
- Title default becomes `chore: release v0.24.0` (new `${tag}` template variable; `${count}` now excludes the root). The Zod default for `standingPr.title` is removed so the strategy-dependent default resolves at use site — configured titles behave exactly as before.
- Body leads with the version and lists bare package names; the root row is excluded from the publish table in all modes (it is never published).

**Thresholds (behavior change):** `standingPr.minPackages` and `ci.minChanges` now count publishable packages only — previously the root bump inflated the count by one in sync mode.

Also includes two lint hygiene commits: pre-existing biome 2.4.16 format drift fixed across tsconfigs/fixtures, and the root `lint` script now runs the full `biome check` (formatter included) — previously CI lint disabled the formatter, so format drift was never caught and `turbo run lint` could fail locally on commits CI had passed.

## Testing

- New unit specs for `version-display.ts` (root filtering, shared-tag display, version range)
- Sync rendering + legacy-manifest fallback cases in `preview-format.spec.ts`
- Sync title default, root-excluded `${count}`, and sync PR body in `standing-pr.spec.ts`
- JSON output contract tests for `strategy` / `isRoot`
- Full workspace `lint + typecheck + test` green (24 tasks)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<details open><summary><h3>Greptile Summary</h3></summary>

This PR introduces sync-aware rendering for the release preview comment and standing PR by recording `strategy` and `isRoot` on `VersionOutput`. Both fields are optional for backwards compatibility with existing manifests.

- **`VersionOutput.strategy`** (`'sync' | 'single' | 'async'`) — set at the start of each versioning strategy function; lets preview/standing-PR logic render sync releases as a single versioned unit rather than a misleading package count.
- **`VersionPackageUpdate.isRoot`** — marks the workspace-root lockstep bump so it can be excluded from package counts, PR titles, and publish tables; `publishableUpdates()` provides a centralized filter with a fallback-to-all safety net for old manifests.
- **Threshold fixes** — `ci.minChanges` and `standingPr.minPackages` now count publishable packages only; the standing-PR title gains a `${tag}` variable and the sync default is `chore: release ${tag}` while the `title` Zod default is removed so the strategy-dependent default is resolved at use-time.
</details>

<h3>Confidence Score: 5/5</h3>

Safe to merge — the changes are additive, both new fields are optional, and all existing code paths have a backward-compatible fallback.

All new VersionOutput fields are optional, so old manifests fall through cleanly to the pre-existing rendering. The publishableUpdates helper's fallback-to-all guard prevents any zero-count edge case from surfacing. Threshold logic, title generation, and body rendering each have dedicated new tests.

No files require special attention.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/release/src/version-display.ts | New module providing `publishableUpdates`, `syncVersionDisplay`, and `syncVersionRange` — clean implementations with correct fallback-to-all edge-case handling. |
| packages/release/src/preview/format.ts | Sync branch now shows the version range in the summary and lists bare package names under a heading; non-sync path and legacy manifests fall through to the old rendering. |
| packages/release/src/standing-pr/standing-pr.ts | PR body, title, and threshold all updated to use `publishableUpdates`; the Zod default for title is removed and the strategy-dependent default is resolved at use-time. |
| packages/core/src/types.ts | Adds optional `strategy` to `VersionOutput` and optional `isRoot` to `VersionPackageUpdate`; both optional preserves backward compatibility. |
| packages/version/src/utils/jsonOutput.ts | Adds `setVersioningStrategy` and `isRoot` spreading in `addPackageUpdate`; `isRoot` is omitted rather than set to false to keep the JSON shape clean for existing consumers. |
| packages/version/src/core/versionStrategies.ts | Calls `setVersioningStrategy` at the top of each strategy and marks the root update with `isRoot: true` in sync mode. |
| packages/config/src/schema.ts | Removes the `title` Zod default so the strategy-dependent default is resolved at runtime; callers handle `undefined` with `?? defaultTitle`. |
| packages/release/test/unit/version-display.spec.ts | New unit tests covering root filtering, shared-tag display, per-package tag fallback, and version range edge cases. |
| packages/release/test/unit/preview-format.spec.ts | New sync-mode specs including the legacy-manifest fallback for both the snapshot line and the details summary. |
| packages/release/test/unit/standing-pr.spec.ts | New tests for sync title default, root-excluded count, and sync PR body format. |

</details>

</details>

<details><summary><h3>Flowchart</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[VersionOutput] -->|strategy === sync| B[Sync Path]
    A -->|strategy !== sync or undefined| C[Legacy/Async Path]
    B --> D[publishableUpdates - filter isRoot]
    D --> E[syncVersionDisplay]
    E --> F[syncVersionRange prev to next]
    B --> G[Snapshot: version queued N packages]
    B --> H[PR body: list bare names]
    B --> I[PR title: chore release tag]
    B --> J[Thresholds: publishable count only]
    C --> K[Snapshot: N packages queued]
    C --> L[PR body: Package/Version table]
    C --> M[PR title: chore release count]
```
</details>

<details><summary><h3>Comments Outside Diff (1)</h3></summary>

1. `packages/release/src/preview/format.ts`, line 388-400 ([link](https://github.com/goosewobbler/releasekit/blob/c18441bd1f233e124d7e53b7a1ebd6104c4c2e3e/packages/release/src/preview/format.ts#L388-L400)) 

   <a href="#">P2</a> **Inline root filter diverges from `publishableUpdates` fallback**

   `renderStandingPRSnapshot` computes the count with `versionOutput.updates.filter((u) => !u.isRoot).length` directly, which returns 0 when all entries are marked root. `publishableUpdates()` (used everywhere else) falls back to all updates in that case, keeping counts consistent. In the hypothetical edge case where only the root update survives, the snapshot would show `v1.2.3 queued` (no count) while the PR body and title would show 1 package. The divergence is unlikely in practice but the intent of `publishableUpdates` is precisely to handle this consistently.

   <a href="https://app.greptile.com/ide/claude-code?prompt=This%20is%20a%20comment%20left%20during%20a%20code%20review.%0APath%3A%20packages%2Frelease%2Fsrc%2Fpreview%2Fformat.ts%0ALine%3A%20388-400%0A%0AComment%3A%0A**Inline%20root%20filter%20diverges%20from%20%60publishableUpdates%60%20fallback**%0A%0A%60renderStandingPRSnapshot%60%20computes%20the%20count%20with%20%60versionOutput.updates.filter%28%28u%29%20%3D%3E%20!u.isRoot%29.length%60%20directly%2C%20which%20returns%200%20when%20all%20entries%20are%20marked%20root.%20%60publishableUpdates%28%29%60%20%28used%20everywhere%20else%29%20falls%20back%20to%20all%20updates%20in%20that%20case%2C%20keeping%20counts%20consistent.%20In%20the%20hypothetical%20edge%20case%20where%20only%20the%20root%20update%20survives%2C%20the%20snapshot%20would%20show%20%60v1.2.3%20queued%60%20%28no%20count%29%20while%20the%20PR%20body%20and%20title%20would%20show%201%20package.%20The%20divergence%20is%20unlikely%20in%20practice%20but%20the%20intent%20of%20%60publishableUpdates%60%20is%20precisely%20to%20handle%20this%20consistently.%0A%0AHow%20can%20I%20resolve%20this%3F%20If%20you%20propose%20a%20fix%2C%20please%20make%20it%20concise.&repo=goosewobbler%2Freleasekit&pr=242&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInClaudeDark.svg?v=3"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInClaude.svg?v=3">Fix in Claude Code</picture></a> <a href="https://app.greptile.com/api/ide/cursor?prompt=This%20is%20a%20comment%20left%20during%20a%20code%20review.%0APath%3A%20packages%2Frelease%2Fsrc%2Fpreview%2Fformat.ts%0ALine%3A%20388-400%0A%0AComment%3A%0A**Inline%20root%20filter%20diverges%20from%20%60publishableUpdates%60%20fallback**%0A%0A%60renderStandingPRSnapshot%60%20computes%20the%20count%20with%20%60versionOutput.updates.filter%28%28u%29%20%3D%3E%20!u.isRoot%29.length%60%20directly%2C%20which%20returns%200%20when%20all%20entries%20are%20marked%20root.%20%60publishableUpdates%28%29%60%20%28used%20everywhere%20else%29%20falls%20back%20to%20all%20updates%20in%20that%20case%2C%20keeping%20counts%20consistent.%20In%20the%20hypothetical%20edge%20case%20where%20only%20the%20root%20update%20survives%2C%20the%20snapshot%20would%20show%20%60v1.2.3%20queued%60%20%28no%20count%29%20while%20the%20PR%20body%20and%20title%20would%20show%201%20package.%20The%20divergence%20is%20unlikely%20in%20practice%20but%20the%20intent%20of%20%60publishableUpdates%60%20is%20precisely%20to%20handle%20this%20consistently.%0A%0AHow%20can%20I%20resolve%20this%3F%20If%20you%20propose%20a%20fix%2C%20please%20make%20it%20concise.&pr=242&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInCursorDark.svg?v=3"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInCursor.svg?v=3">Fix in Cursor</picture></a>
</details>

<sub>Reviews (2): Last reviewed commit: ["fix(release): address review findings on..."](https://github.com/goosewobbler/releasekit/commit/5af1aa028c595794c6e65d6c134fb4c6dea1933e) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=35598088)</sub>

### CODE REVIEW

File: packages/release/src/preview/format.ts [hunk 1]
```diff
@@ -333,15 +354,28 @@ export function formatPreviewComment(result: ReleaseOutput | null, options?: For
...
+  let queuedSummary: string;
+  if (versionOutput.strategy === 'sync') {
+    // Sync mode emits a single aggregated changelog, so counting changelogs would always say
+    // "1 package" — lead with the queued version instead, with the real package count
+    // (excluding the root lockstep bump) as detail.
+    const pkgCount = versionOutput.updates.filter((u) => !u.isRoot).length;
+    const pkgPart = pkgCount > 0 ? ` (${pkgCount} ${pkgCount === 1 ? 'package' : 'packages'})` : '';
+    queuedSummary = `${syncVersionDisplay(versionOutput)} queued${pkgPart}`;
+  } else {
+    const pkgCount = versionOutput.changelogs.filter(
```
[the remark this answers is not in the document]
Comment (goosewobbler) [reply packages/release/src/preview/format.ts:375]: Fixed in 5af1aa0 — the heading now claims "All packages" only when the list covers every publishable package, falling back to "Also bumped → <version>" when packages with their own changelog entries render above it. Test added for the subset case.
